twisted/test/proto_helpers.py

```diff
@@ -142,6 +142,13 @@ class StringTransport:
     @ivar io: A L{io.BytesIO} which holds the data which has been written to
         this transport since the last call to L{clear}.  Use L{value} instead
         of accessing this directly.
+
+    @ivar lenient: A L{bool} that controls whether L{StringTransport} will
```

Comment:
I know that @glyph has suggested this flag, but I think that we can do better by creating 2 string transports like StateEnforcingStringTransport and RealWorldStringTransport ... note that the names should be improved, but I hope that you get the idea.

In the class docstring we can also include all the comment from the Trac ticket with information about when to use one class and when to use another class.

I find the current docstring confusing as it does not explain why on earth you would like to run your tests with a mock which is not modeled after the real deal :)